src/main/java/org/opensearch/geospatial/processor/GeoJSONFeatureProcessor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.processor;
+
+import org.opensearch.geospatial.geojson.Feature;
+import org.opensearch.geospatial.geojson.GeoJSONFactory;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+
+import java.util.Map;
+
+import static org.opensearch.ingest.ConfigurationUtils.readStringProperty;
+
+public class GeoJSONFeatureProcessor extends AbstractProcessor {
+
+    public static final String FIELD_KEY = "field";
+    public static final String TYPE = "geojson";
+    private final String geoShapeField;
+
+    public GeoJSONFeatureProcessor(String tag, String description, String geoShapeField) {
+        super(tag, description);
+        this.geoShapeField = geoShapeField;
+    }
+
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        Feature feature = GeoJSONFactory.create(ingestDocument.getSourceAndMetadata());
```

Comment:
Can you add comments here explaining what is being done?

Reply:
updated.